python/pyarrow/tests/test_pandas.py

```diff
@@ -3325,13 +3325,35 @@ def test_cast_timestamp_unit():
     assert result.equals(expected)
 
 
-def test_struct_with_timestamp_tz():
+def test_nested_with_timestamp_tz_round_trip():
+    ts = pd.Timestamp.now()
+    ts_dt = ts.to_pydatetime()
+    arr = pa.array([ts_dt], type=pa.timestamp('us', tz='America/New_York'))
+    struct = pa.StructArray.from_arrays([arr, arr], ['start', 'stop'])
+
+    result = struct.to_pandas()
+    # N.B. we test with Pandas because the Arrow types are not
```

Comment:
this might not be true anymore with StringToTz?

Reply:
Indeed it seems like `pa.array(result)` should recover `struct` and we want to verify that?